packages/zoe/src/contractFacet/types-ambient.d.ts

```diff
@@ -246,3 +243,11 @@ type ContractStartFnResult<PF, CF> = {
 };
 type ContractOf<S> = import('../zoeService/utils').ContractOf<S>;
 type AdminFacet = import('../zoeService/utils').AdminFacet<any>;
+
+declare const OfferReturn: unique symbol;
+declare const OfferArgs: unique symbol;
+type Invitation<R = unknown, A = undefined> = Payment<'set'> & {
+  // because TS is structural, without this the generic is ignored
```

Comment:
Still mysterious and weird. But thanks for the comment. Without it I would have been flummoxed. No change suggested.

Reply:
Because its type system is structural, anything that's not part of the object gets elided. This puts properties on the structure of the object that have no risk of conflict because they're unique symbols.

type-fest has this Opaque utility which explains more: https://github.com/sindresorhus/type-fest/blob/main/source/opaque.d.ts

So far we've avoided taking that package as a dependency